src/platforms/hosted/cli.c

```diff
@@ -480,7 +480,10 @@ int cl_execute(bmda_cli_options_s *opt)
 	platform_nrst_set_val(opt->opt_connect_under_reset);
 	if (opt->opt_mode == BMP_MODE_TEST)
 		DEBUG_INFO("Running in Test Mode\n");
-	DEBUG_INFO("Target voltage: %s Volt\n", platform_target_voltage());
+	/* Call the sampling function only once, check for NULL, check for unimplemented and suppress the output */
+	const char *const target_voltage = platform_target_voltage();
+	if (target_voltage && (!strcmp(target_voltage, "Unknown")))
+		DEBUG_INFO("Target voltage: %s Volt\n", target_voltage);
```

Comment:
This gets a similar request as the command.c version, but including the suggestion to let's drop the world "Volt" from here. This has been the source of all of the funky in BMDA that the firmware doesn't have.

It should be up to the `platform_target_voltage()` implementation to include the unit on the end of the formatted string, as in "3.3V" or "0.0V" or w/e. That allows the display of "Target voltage: Unknown" or "Target voltage: Present" etc to be nice and consistent across BMDA and the firmware.

Reply:
> so how about we revisit also the hosted backends?.

Probably a very good idea to check that they all suffix valid voltages with `V` and have sensible invalid voltage displays as that's really part and parcel of fixing this 👍🏼